src/Squirrel/UpdateManager.InstallHelpers.cs

```diff
@@ -93,56 +105,81 @@ public async Task<RegistryKey> CreateUninstallerRegistryEntry(string uninstallCm
                     new { Key = "Language", Value = 0x0409 },
                 };
 
-                foreach (var kvp in stringsToWrite) {
+                foreach (var kvp in stringsToWrite)
+                {
                     key.SetValue(kvp.Key, kvp.Value, RegistryValueKind.String);
                 }
-                foreach (var kvp in dwordsToWrite) {
+                foreach (var kvp in dwordsToWrite)
+                {
                     key.SetValue(kvp.Key, kvp.Value, RegistryValueKind.DWord);
                 }
 
                 return key;
             }
 
-            public void KillAllProcessesBelongingToPackage()
+            public int KillAllProcessesBelongingToPackage()
             {
                 var ourExe = Assembly.GetEntryAssembly();
                 var ourExePath = ourExe != null ? ourExe.Location : null;
 
-                IEnumerable<Tuple<string, int>> running = null;
-                int maxRetries = 3;
-                do
+                List<Tuple<string, int>> running = null;
```

Comment:
[optional] the parent process id may be a part of the tuple. This would avoid the ` Process process = Process.GetProcessById(id)` in GetParentProcessId, as the process reference is already available.
Also, having named tuple members would help the readability.